src/Mod/Sketcher/Gui/SketcherSettings.cpp

```diff
@@ -66,6 +66,42 @@ void SketcherSettings::saveSettings()
     ui->checkBoxEnableEscape->onSave();
     ui->checkBoxNotifyConstraintSubstitutions->onSave();
     ui->checkBoxAutoRemoveRedundants->onSave();
+
+    // Dimensioning constraints mode
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Sketcher/dimensioning");
+    bool singleTool = true;
+    bool SeparatedTools = false;
+    int index = ui->dimensioningMode->currentIndex();
+    switch (index) {
+    case 1:
```

Comment:
Avoid magic numbers and instead choose some speaking enumerators. Same applies below...

Reply:
Please have a look at the commit [79bb68e](https://github.com/FreeCAD/FreeCAD/pull/9810/commits/79bb68ef142f9273753954fd356ec00f3a64a2af) to let me know if this is what you had in mind. I don't see how we can use modern class enum here so I used legacy enums. Let me know.

Reply:
Old-style enums are fine because it anyway is only an implementation detail, i.e. not part of the API this class.